src/history/hs_rec.c

```diff
@@ -620,19 +629,14 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
             }
 
             /* Skip updates that are already in the history store. */
-            if (F_ISSET(upd, WT_UPDATE_HS)) {
+            if (F_ISSET(upd, WT_UPDATE_TO_DELETE_FROM_HS | WT_UPDATE_HS)) {
                 if (hs_inserted)
                     WT_ERR_PANIC(session, WT_PANIC,
                       "Reinserting updates to the history store may corrupt the data as it may "
                       "clear the history store data newer than it.");
                 continue;
             }
 
-            /* We should never write a prepared update to the history store. */
```

Comment:
This assertion is still valid, why is it removed?

Reply:
It is repetitive. We have asserted before.